Cannot install 1.5.0, 1.4.4 works fine.

I'm on a M1 Mac mini with an up to date installation of MacPorts + Ruby 3.1.2.

```
samuel@Motoko ~/P/k/vizor (master✔) [2]> gem install rugged --version 1.4.4
Fetching rugged-1.4.4.gem
Building native extensions. This could take a while...
load: 5.00  cmd: clang 62010 running 0.11u 0.01s
Successfully installed rugged-1.4.4
Parsing documentation for rugged-1.4.4
Installing ri documentation for rugged-1.4.4
Done installing documentation for rugged after 1 seconds
1 gem installed
samuel@Motoko ~/P/k/vizor (master✔)> gem install rugged --version 1.5.0
Building native extensions. This could take a while...
ERROR:  Error installing rugged:
	ERROR: Failed to build gem native extension.

    current directory: /Users/samuel/.gem/ruby/3.1.2/gems/rugged-1.5.0/ext/rugged
/Users/samuel/.rubies/ruby-3.1.2/bin/ruby -I /Users/samuel/.rubies/ruby-3.1.2/lib/ruby/3.1.0 -r ./siteconf20220725-62235-loqfsy.rb extconf.rb
checking for gmake... yes
checking for cmake... yes
checking for pkg-config... yes
-- The C compiler identification is AppleClang 13.1.6.13160021
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /opt/local/bin/pkg-config (found version "0.29.2") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM - Failed
-- Performing Test HAVE_STRUCT_STAT_ST_MTIMESPEC
-- Performing Test HAVE_STRUCT_STAT_ST_MTIMESPEC - Success
-- Performing Test HAVE_STRUCT_STAT_MTIME_NSEC
-- Performing Test HAVE_STRUCT_STAT_MTIME_NSEC - Failed
-- Performing Test HAVE_STRUCT_STAT_NSEC
-- Performing Test HAVE_STRUCT_STAT_NSEC - Success
-- Performing Test IS_WERROR_SUPPORTED
-- Performing Test IS_WERROR_SUPPORTED - Success
-- Looking for futimens
-- Looking for futimens - found
-- Performing Test IS_WNO_ERROR_SUPPORTED
-- Performing Test IS_WNO_ERROR_SUPPORTED - Success
-- Performing Test IS_WALL_SUPPORTED
-- Performing Test IS_WALL_SUPPORTED - Success
-- Performing Test IS_WEXTRA_SUPPORTED
-- Performing Test IS_WEXTRA_SUPPORTED - Success
-- Performing Test IS_WDOCUMENTATION_SUPPORTED
-- Performing Test IS_WDOCUMENTATION_SUPPORTED - Success
-- Performing Test IS_WNO_DOCUMENTATION_DEPRECATED_SYNC_SUPPORTED
-- Performing Test IS_WNO_DOCUMENTATION_DEPRECATED_SYNC_SUPPORTED - Success
-- Performing Test IS_WNO_MISSING_FIELD_INITIALIZERS_SUPPORTED
-- Performing Test IS_WNO_MISSING_FIELD_INITIALIZERS_SUPPORTED - Success
-- Performing Test IS_WMISSING_DECLARATIONS_SUPPORTED
-- Performing Test IS_WMISSING_DECLARATIONS_SUPPORTED - Success
-- Performing Test IS_WSTRICT_ALIASING_SUPPORTED
-- Performing Test IS_WSTRICT_ALIASING_SUPPORTED - Success
-- Performing Test IS_WSTRICT_PROTOTYPES_SUPPORTED
-- Performing Test IS_WSTRICT_PROTOTYPES_SUPPORTED - Success
-- Performing Test IS_WDECLARATION_AFTER_STATEMENT_SUPPORTED
-- Performing Test IS_WDECLARATION_AFTER_STATEMENT_SUPPORTED - Success
-- Performing Test IS_WSHIFT_COUNT_OVERFLOW_SUPPORTED
-- Performing Test IS_WSHIFT_COUNT_OVERFLOW_SUPPORTED - Success
-- Performing Test IS_WUNUSED_CONST_VARIABLE_SUPPORTED
-- Performing Test IS_WUNUSED_CONST_VARIABLE_SUPPORTED - Success
-- Performing Test IS_WUNUSED_FUNCTION_SUPPORTED
-- Performing Test IS_WUNUSED_FUNCTION_SUPPORTED - Success
-- Performing Test IS_WINT_CONVERSION_SUPPORTED
-- Performing Test IS_WINT_CONVERSION_SUPPORTED - Success
-- Performing Test IS_WC11_EXTENSIONS_SUPPORTED
-- Performing Test IS_WC11_EXTENSIONS_SUPPORTED - Success
-- Performing Test IS_WC99_C11_COMPAT_SUPPORTED
-- Performing Test IS_WC99_C11_COMPAT_SUPPORTED - Failed
-- Performing Test IS_WFORMAT_SUPPORTED
-- Performing Test IS_WFORMAT_SUPPORTED - Success
-- Performing Test IS_WFORMAT_SECURITY_SUPPORTED
-- Performing Test IS_WFORMAT_SECURITY_SUPPORTED - Success
-- Found GSSAPI: /opt/local/lib/libgssapi_krb5.dylib;/opt/local/lib/libkrb5.dylib;/opt/local/lib/libk5crypto.dylib;/opt/local/lib/libcom_err.dylib  
-- Found GSS.framework /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/GSS.framework
-- Found OpenSSL: /opt/local/libexec/openssl3/lib/libcrypto.dylib (found version "3.0.5")  
-- Found Security /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/Security.framework
-- Looking for SSLCreateContext in /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/Security.framework
-- Looking for SSLCreateContext in /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/Security.framework - found
-- Found CoreFoundation /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/CoreFoundation.framework
-- http-parser version 2 was not found or disabled; using bundled 3rd-party sources.
-- Performing Test IS_WIMPLICIT_FALLTHROUGH_1_SUPPORTED
-- Performing Test IS_WIMPLICIT_FALLTHROUGH_1_SUPPORTED - Failed
-- Found PCRE: /opt/local/lib/libpcre.dylib  
-- Looking for dirent.h
-- Looking for dirent.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for windows.h
-- Looking for windows.h - not found
-- Looking for bcopy
-- Looking for bcopy - found
-- Looking for memmove
-- Looking for memmove - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for strtoll
-- Looking for strtoll - found
-- Looking for strtoq
-- Looking for strtoq - found
-- Looking for _strtoi64
-- Looking for _strtoi64 - not found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of long long
-- Check size of long long - done
-- Check size of unsigned long long
-- Check size of unsigned long long - done
-- Performing Test IS_WNO_UNUSED_FUNCTION_SUPPORTED
-- Performing Test IS_WNO_UNUSED_FUNCTION_SUPPORTED - Success
-- Performing Test IS_WNO_IMPLICIT_FALLTHROUGH_SUPPORTED
-- Performing Test IS_WNO_IMPLICIT_FALLTHROUGH_SUPPORTED - Success
-- LIBSSH2 not found. Set CMAKE_PREFIX_PATH if it is installed outside of the default search path.
-- Found ZLIB: /opt/local/lib/libz.dylib (found version "1.2.12") 
-- Checking prototype qsort_r for GIT_QSORT_R_BSD
-- Checking prototype qsort_r for GIT_QSORT_R_BSD - True
-- Checking prototype qsort_r for GIT_QSORT_R_GNU
-- Checking prototype qsort_r for GIT_QSORT_R_GNU - False
-- Looking for qsort_s
-- Looking for qsort_s - not found
-- Looking for getentropy
-- Looking for getentropy - found
-- Looking for getloadavg
-- Looking for getloadavg - found
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - not found
-- Looking for iconv_open
-- Looking for iconv_open - not found
-- Found Iconv: -L/opt/local/lib -liconv
-- Enabled features:
 * nanoseconds, support nanosecond precision file mtimes and ctimes
 * HTTPS, using SecureTransport
 * SHA1, using CollisionDetection
 * SHA256, using CommonCrypto
 * http-parser, http-parser support (bundled)
 * regex, using bundled PCRE
 * zlib, using system zlib
 * futimens, futimens support
 * threadsafe, threadsafe support
 * ntlmclient, NTLM authentication support for Unix
 * iconv, iconv encoding conversion support

-- Disabled features:
 * debugpool, debug pool allocator
 * debugalloc, debug strict allocators
 * debugopen, path validation in open
 * SPNEGO, SPNEGO authentication support
 * SSH, SSH transport support

-- Configuring done
-- Generating done
-- Build files have been written to: /Users/samuel/.gem/ruby/3.1.2/gems/rugged-1.5.0/vendor/libgit2/build
 -- /opt/local/bin/gmake
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/Users/samuel/.rubies/ruby-3.1.2/bin/$(RUBY_BASE_NAME)
	--with-sha1dc
	--without-sha1dc
	--with-ssh
	--without-ssh
	--use-system-libraries
extconf.rb:24:in `sys': ERROR: '/opt/local/bin/gmake' failed (RuntimeError)
	from extconf.rb:108:in `block (2 levels) in <main>'
	from extconf.rb:104:in `chdir'
	from extconf.rb:104:in `block in <main>'
	from extconf.rb:101:in `chdir'
	from extconf.rb:101:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /Users/samuel/.gem/ruby/3.1.2/extensions/arm64-darwin-21/3.1.0-static/rugged-1.5.0/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /Users/samuel/.gem/ruby/3.1.2/gems/rugged-1.5.0 for inspection.
Results logged to /Users/samuel/.gem/ruby/3.1.2/extensions/arm64-darwin-21/3.1.0-static/rugged-1.5.0/gem_make.out
```

I've attach `mkmf.log` and `gem_make.out` in a zip file: 
[rugged-1.5.0.zip](https://github.com/libgit2/rugged/files/9177340/rugged-1.5.0.zip)

Please let me know how I can help you triage/fix this, if you need any more information, want me to try things, etc.

```
Undefined symbols for architecture arm64:
  "_iconv", referenced from:
      _unicode_iconv_encoding_convert in unicode_iconv.c.o
     (maybe you meant: _git_fs_path_iconv_init_precompose, _git_fs_path_iconv_clear , _git_fs_path_iconv )
  "_iconv_close", referenced from:
      _ntlm_unicode_shutdown in unicode_iconv.c.o
  "_iconv_open", referenced from:
      _ntlm_unicode_init in unicode_iconv.c.o
ld: symbol(s) not found for architecture arm64
```

This sounds like a bug in libgit2's geration of the pc file which is what we use to know what to link to.

--- a/ext/rugged/extconf.rb
+++ b/ext/rugged/extconf.rb
@@ -14,6 +14,8 @@ $CFLAGS << " -O3" unless $CFLAGS[/-O\d/]
 $CFLAGS << " -Wall -Wno-comment"
 
 cmake_flags = [ ENV["CMAKE_FLAGS"] ]
+cmake_flags << "-DBUILD_CLI=OFF"
+cmake_flags << "-DBUILD_TESTS=OFF"
 cmake_flags << "-DREGEX_BACKEND=builtin"
 cmake_flags << "-DUSE_SHA1DC=ON" if with_config("sha1dc")
 cmake_flags << "-DUSE_SSH=ON"    if with_config("ssh")

